### PR TITLE
Np 32272 search publisher

### PR DESCRIPTION
**Pull request by hackon**

### CODE REVIEW

File: docs/open-api.yaml [hunk 1]
```diff
@@ -380,7 +500,28 @@ components:
...
+          description: The uri to the next page of results, if there is one
+        previousResults:
+          type: string
+          nullable: true
+          description: The uri to the previous page of results, if there is one
+        hits:
+          type: array
+          items:
+            $ref: '#/components/schemas/Publisher'
+          description: The url to the publisher
```
Comment (monaullah) [inline docs/open-api.yaml:524]: Fix description for hits

File: docs/open-api.yaml [hunk 2]
```diff
@@ -427,6 +568,53 @@ components:
...
+          description: The uri to the next page of results, if there is one
+        previousResults:
+          type: string
+          nullable: true
+          description: The uri to the previous page of results, if there is one
+        hits:
+          type: array
+          items:
+            $ref: '#/components/schemas/Series'
+          description: The url to the publisher
```
Comment (monaullah) [inline docs/open-api.yaml:592]: Fix description for hits

File: template.yaml [hunk 3]
```diff
@@ -188,6 +188,27 @@ Resources:
       Action: lambda:InvokeFunction
       Principal: 'apigateway.amazonaws.com'
 
+  SearchPublisherByQueryFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Handler: no.sikt.nva.pubchannels.handler.search.journal.SearchPublisherByQueryHandler::handleRequest
```
Comment (monaullah) [inline template.yaml:194]: no.sikt.nva.pubchannels.handler.search.publisher.

File: template.yaml [hunk 4]
```diff
@@ -211,6 +232,27 @@ Resources:
       Action: lambda:InvokeFunction
       Principal: 'apigateway.amazonaws.com'
 
+  SearchSeriesByQueryFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Handler: no.sikt.nva.pubchannels.handler.search.journal.SearchSeriesByQueryHandler::handleRequest
```
Comment (monaullah) [inline template.yaml:238]: no.sikt.nva.pubchannels.handler.search.series.

File: src/main/java/no/sikt/nva/pubchannels/dataporten/DataportenPublicationChannelClient.java [hunk 5]
```diff
@@ -69,10 +69,10 @@ public ThirdPartyPublicationChannel getChannel(ChannelType type, String identifi
     }
 
     @Override
-    public DataportenSearchJournalResponse getChannel(ChannelType type, Map<String, String> queryParameters)
+    public DataportenSearchResponse getChannel(ChannelType type, Map<String, String> queryParameters)
```
Comment (monaullah) [inline src/main/java/no/sikt/nva/pubchannels/dataporten/DataportenPublicationChannelClient.java:72]: Separate response dtos?

File: src/main/java/no/sikt/nva/pubchannels/handler/search/journal/JournalResult.java [hunk 6]
```diff
@@ -54,7 +54,7 @@ public JournalResult(@JsonProperty(ID_FIELD) URI id,
         this.sameAs = sameAs;
     }
 
-    public static JournalResult create(URI selfUriBase, DataportenJournalResult journal) {
+    public static JournalResult create(URI selfUriBase, DataportenEntityResult journal) {
```
Comment (monaullah) [inline src/main/java/no/sikt/nva/pubchannels/handler/search/journal/JournalResult.java:57]: Make constructor private?

File: src/test/java/no/sikt/nva/pubchannels/handler/search/journal/SearchJournalByQueryHandlerTest.java [hunk 7]
```diff
@@ -64,20 +76,10 @@
...
-    public static final String ISSN_QUERY_PARAM = "issn";
-    public static final String PID_QUERY_PARAM = "pid";
-    public static final int MAX_LEVEL = 2;
-    public static final double MIN_LEVEL = 0;
-    public static final String DEFAULT_OFFSET = "0";
-    public static final String DEFAULT_SIZE = "10";
     private static final Context context = new FakeContext();
-    private static final String DATAPORTEN_PAGE_NO_PARAM = "pageno";
-    private static final String DATAPORTEN_PAGE_COUNT_PARAM = "pagecount";
+    public static final TypeReference<PaginatedSearchResult<JournalResult>> TYPE_REF = new TypeReference<>() {
```
Comment (monaullah) [inline src/test/java/no/sikt/nva/pubchannels/handler/search/journal/SearchJournalByQueryHandlerTest.java:81]: private

File: src/test/java/no/sikt/nva/pubchannels/handler/search/series/SearchSeriesByQueryHandlerTest.java [hunk 8]
```diff
@@ -0,0 +1,500 @@
...
+            String printIssn)
+            throws UnprocessableContentException {
+        var pid = UUID.randomUUID().toString();
+        var name = randomString();
+        var electronicIssn = randomIssn();
+        var level = randomLevel();
+        var landingPage = randomUri();
+
+        List<DataportenEntityResult> dataportenEntityResult = List.of(
+                createDataportenJournalResult(year, printIssn, pid, name, electronicIssn, landingPage, level)
```
Comment (monaullah) [inline src/test/java/no/sikt/nva/pubchannels/handler/search/series/SearchSeriesByQueryHandlerTest.java:327]: createDataportenChannelResult?

File: src/test/java/no/sikt/nva/pubchannels/handler/search/series/SearchSeriesByQueryHandlerTest.java [hunk 9]
```diff
@@ -0,0 +1,500 @@
...
+    }
+
+    private PaginatedSearchResult<SeriesResult> getExpectedPaginatedSearchResultPidSearch(String year, String pid)
+            throws UnprocessableContentException {
+        var printIssn = randomIssn();
+        var name = randomString();
+        var electronicIssn = randomIssn();
+        var level = randomLevel();
+        var landingPage = randomUri();
+        var dataportenJournalResult = List.of(
```
Comment (monaullah) [inline src/test/java/no/sikt/nva/pubchannels/handler/search/series/SearchSeriesByQueryHandlerTest.java:347]: dataportenSeriesResult

File: src/test/java/no/sikt/nva/pubchannels/handler/search/series/SearchSeriesByQueryHandlerTest.java [hunk 10]
```diff
@@ -0,0 +1,500 @@
...
+                .collect(Collectors.toList());
+    }
+
+    private PaginatedSearchResult<SeriesResult> getExpectedPaginatedSearchResultNameSearch(
+            List<DataportenEntityResult> dataportenResults,
+            String year,
+            String name, int queryOffset, int querySize) throws UnprocessableContentException {
+        var expectedHits = dataportenResults
+                .stream()
+                .map(this::toJournal)
```
Comment (monaullah) [inline src/test/java/no/sikt/nva/pubchannels/handler/search/series/SearchSeriesByQueryHandlerTest.java:481]: toSeries

File: src/test/java/no/sikt/nva/pubchannels/handler/search/publisher/SearchPublisherByQueryHandlerTest.java [hunk 11]
```diff
@@ -0,0 +1,504 @@
...
+                .collect(Collectors.toList());
+    }
+
+    private PaginatedSearchResult<PublisherResult> getExpectedPaginatedSearchResultNameSearch(
+            List<DataportenEntityResult> dataportenResults,
+            String year,
+            String name, int queryOffset, int querySize) throws UnprocessableContentException {
+        var expectedHits = dataportenResults
+                .stream()
+                .map(this::toJournal)
```
Comment (monaullah) [inline src/test/java/no/sikt/nva/pubchannels/handler/search/publisher/SearchPublisherByQueryHandlerTest.java:485]: toPublisher

**Review by monaullah — APPROVED** [review]

File: src/main/java/no/sikt/nva/pubchannels/dataporten/DataportenPublicationChannelClient.java (hunk 5 above)
Comment (hackon) [reply src/main/java/no/sikt/nva/pubchannels/dataporten/DataportenPublicationChannelClient.java:72]: Will not do now.

**Pull request merged**